filtered-resource-service/src/main/java/uk/gov/gchq/palisade/service/filteredresource/stream/config/AkkaRunnableGraph.java

```diff
@@ -177,41 +191,92 @@ FilteredResourceSourceFactory filteredResourceSourceFactory(final PartitionedOff
+                                            .orElse(!observedStart.get()))
+
+                                    .map((Tuple4<String, Optional<StreamMarker>, Optional<FilteredResourceRequest>, CommittableMessage<String, FilteredResourceRequest>> tokenMarkerRequestCommittable) -> {
+                                        // Only audit once, not for each resource in which the Start Marker is missing
+                                        observedStart.set(true);
+
+                                        // Extract the headers
+                                        Headers headers = tokenMarkerRequestCommittable.t4().record().headers();
+                                        // Extract the FilteredResourceRequest from the optional
+                                        return tokenMarkerRequestCommittable.t3()
+                                                .map((FilteredResourceRequest filteredResourceRequest) -> {
+
+                                                    // Build the error message
+                                                    AuditErrorMessage auditErrorMessage = AuditErrorMessage.Builder.create().withUserId(filteredResourceRequest.getUserId())
+                                                            .withResourceId(filteredResourceRequest.getResourceId())
+                                                            .withContextNode(filteredResourceRequest.getContextNode())
+                                                            .withAttributes(Collections.emptyMap())
+                                                            .withError(new NoStartMarkerObservedException("No Start Marker was observed for token: " + tokenMarkerRequestCommittable.t1()));
+
+                                                    // Create the ProducerRecord, on the error topic, on the right partition, with the audit error message
+                                                    LOGGER.debug("NoStartMarkerObservedException thrown for token {}, on partition {} and topic {}", tokenMarkerRequestCommittable.t1(), partition, errorTopic.getName());
+                                                    return new ProducerRecord<>(errorTopic.getName(), partition, (String) null, auditErrorMessage, headers);
+                                                });
+                                    })
+
+                                    // If we reach this error state with an end message, just ignore it
+                                    .filter(Optional::isPresent)
+                                    .map(Optional::get)
+
+                                    // Limit the flow to run once
+                                    // Audit the error
+                                    .alsoTo(auditErrorSink)
+                                    .to(Sink.foreach((ProducerRecord<String, AuditErrorMessage> x) -> {
+                                        throw new RuntimeException(x.value().getError().getMessage());
```

Comment:
Why not just throw the the `.getError()` exception?
```suggestion
                                        throw x.value().getError();
```

Reply:
updated